---
title: "Rmarkdown exercise"
author: "Pol"
date: "2023-05-08"
output:
  #pdf_document: default
  html_document:
    df_print: paged
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Available books links
[Intro2R](intro2r.com)

[Rmarkdown, the definitive guide](https://bookdown.org/yihui/rmarkdown/)


# Rmarkdown exercise

#### Packages for Rmarkdown

```{r, eval=FALSE}
install.packages("rmarkdown")
install.packages("tinytex")
tinytex::install_tinytex() # Tinytex is requred for PDF format
library(rmarkdown)
```
#### Create, list and extract directories and files

```{r, eval=FALSE}
dir.create("output") # Create directory
dir.create("./output/figures") 
unzip("all_data.zip") # Unzip ZIP file
list.files() # List files on current directory
list.dirs() # List directories on current directory
```

#### Import data and visualize data structure

I am familiar with these sort of commands and I use them often on my current PhD project. For this reason I do not go into detail.

```{r}
pathways <- read.table(file = "datasets/dataset_overview_pathways.txt", sep = "\t", header = T, stringsAsFactors = T)
```


``summary()`` returns some measures of central tendency of the variables of the data set, whereas ``str()`` only returns the column names and their data type.


```{r}
str(pathways)
```
```{r}
summary(pathways)
```

#### Rmarkdown language

* ``< Ctr + Alt + I >`` opens a chunk of code
* ``echo = FALSE`` can be added next to the ``{r}`` at the chunks of code that do not need to be shown.
* ``eval = FALSE`` can be added next to the ``{r}`` at the chunks of code that do not need to be run at knitting.
* Introducing images: ``![title](path){width=???px, height=???px}``

# Create Version Control
To create a Version Control on Git, a GitHub account is required.

##### Download
Git needs to be installed in the computer. To check if it is installed, the following can be typed in the R terminal: ``git --version``. [Here](https://git-scm.com/downloads) is the link for downloading it.

##### Configure
###### Git in the PC
In the terminal of the computer, this needs to be typed:
```
git config --global user.email 'you@youremail.com'
git config --global user.name 'Your Name'
git config --global --list # Confirm configuration
```
###### Git in RStudio
For RStudio to access Git: ``Tools > Global Options > Git/SVN > Git executable > <path_to_git>``

![Git in RStudio](Figures/Git_in_RStudio.png){width=400px, height=400px}

###### Setting up a project in RStudio
1. Create a new repository in GitHub with a README file
2. Create a new project with Version Control in RStudio
  * Include the URL of the GitHub repository
  
##### Using Git
Four steps on saving the changes:

**Save** in local computer (normally)

**Stage** the file in Git (brak point in the progress)

**Commit** creates a permanent snapshot of changes

**Push** to upload on GitHub

**Pull** to import from GitHub to local computer



# Exploring data and summary statistics

#### **Violin plot: Nucleotide Diversity of all genes**

```{r, message=FALSE, warning=FALSE}
library("ggplot2", verbose = F)
library("dplyr", verbose = F)
# Load data
pathgen <- read.csv("datasets/dataset_overview_genes.txt", sep = "\t", header = T)
# Select column to plot
colnames(pathgen)
column_pl <- "nucl_div"
# Group by pathway
pg_by_pway <- group_by(pathgen, pathway)
# Remove NAs for the column of interest
pg_na <- pg_by_pway[!(is.na(paste("dfg$", column_pl, sep = ""))), ]
# Violin plot variable
colors_pways <- c("STEP" = "#F8766D", "DOPP" = "#7CAE00", "VT-R" = "#00BFC4", "SERP" = "#C77CFF", "OXTP" = "#619CFF") # Assign a color to each pathway
plot_pi <- ggplot(pg_na, aes(x=pathway, y=nucl_div, fill=pathway)) +
  geom_point(aes(color=pathway), position=position_jitter(width=0.1), size=0.5, alpha=0.6, stroke = 0.9, color = "black") +
  scale_fill_manual(values = colors_pways, ) +
  scale_color_manual(values = colors_pways) +
  geom_violin(alpha = 0.4, color = "transparent")+
  geom_text(data = subset(pg_na, nucl_div > 0.018), aes(label = gene), vjust = -1, size = 2) + # Label filter points
  theme_minimal()+
  labs(x="Pathway", y="index"~pi*"")+
  ggtitle("Nucleotide diversity ("~pi*")")
# Display plot
plot_pi
# Summary statistics of each pathway
by(pg_na$nucl_div, pg_na$pathway, summary)
```

#### **Lollipop chart: Top 10 genes of the Oxytocin Signalling Pathway**

```{r, message=FALSE, warning=FALSE}
library("ggplot2", verbose = F)
library("dplyr", verbose = F)
df_top10 <- read.csv("datasets/oxtp_FEL_dNdS.txt", header = T, sep = "\t")
#Rename columns and select the ones required
colnames(df_top10) <- c("gene", "dNdS", "pos_sites", "total_sites", "site_density")
df_top10_f <-df_top10[1:10,]
df_top10_f <- select(df_top10_f, c(gene, dNdS, site_density))
df_top10_f$site_density <- round(df_top10_f$site_density, round(2)) # Round to 2 decimals

#Order by w value
df_s <- df_top10_f[order(df_top10_f$dNdS, decreasing = T), ]
gene_order = c(df_s$gene)
df_top10_f$gene <- factor(df_top10_f$gene, levels = rev(gene_order))

#Plot
dNdS_top10 <- ggplot(df_top10_f, aes(x = gene, y = dNdS)) +
  geom_segment(aes(x = gene, xend = gene, y = 0.2, yend = dNdS), linetype=2, color = "black", linewidth = 0.6) +
  geom_point(aes(size = site_density), fill = "gray", color = "black", shape = 21, alpha = 1, stroke = 1) +
  geom_text(aes(label = paste(site_density, " %", sep = "")), size = 3, color = "black", hjust = -0.5) +
  scale_size(range = c(1, 10)) +
  labs(title = "Top 10 OXTP genes dNdS ratio. ",
       x = "Gene",
       y = "dN/dS",
       size = "Positive sites\ndensity\n(% of sites)") +
  ylim(0.3, 0.7) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 20, face = "bold"),
        axis.title = element_text(size = 12),
        axis.text = element_text(size = 10),
        legend.title = element_text(size = 10),
        legend.text = element_text(size = 10)) + 
  coord_flip()

dNdS_top10
# Summary
summary(df_top10_f)
```

#### **Scatterplot for discrete variables: Pair-bonding vs SNP (ref/alternative)**

```{r, message=FALSE, warning=FALSE}
library("ggplot2", verbose = F)
library("dplyr", verbose = F)

df <- read.csv("datasets/EGFR_19907494_PB_BT.txt", sep = "\t")
df2 <- read.csv("datasets/spp-tribe.txt", sep = "\t")

colnames(df) <- c("spp", "PB", "SNP")

# Merge the 2 data frames to link species and tribes
df_final <- inner_join(df, df2, by="spp")

#Plot
SNP_vs_PB <- ggplot(df_final, aes(x = PB, y = SNP, color = tribe)) +
  geom_point(position=position_jitter(width=0.2, height = 0.2)) +
  scale_x_continuous(breaks = c(0, 1), limits = c(-0.2, 1.2)) +
  scale_y_continuous(breaks = c(0, 1), limits = c(-0.2, 1.2)) +
  labs(title = "SNP allele vs phenotype (Pair-bonding)",
       x = "Pair-bonding (0 = No ; 1 = Yes)",
       y = "SNP (0 = Ref ; 1 = Alt)")

SNP_vs_PB
```

# Inferential Statistics

#### Corelation between nucleotide diversity (π) and dNdS ratio (ω)
We are asking if the nucleotide diversity (index that indicates the level of variability of a nucleotide sequences across a set of species) is corelated with the index of positive selection (ω). By taking the Oxytocin Signalling Pathway, we compute a Spearman's corelation test, which does NOT assume linearity but only monotony (always same sense -positive or negative-)

```{r, message=FALSE, warning=FALSE}
library("ggplot2", verbose = F)
library("dplyr", verbose = F)
# Load data
dataset <- read.csv("datasets/dataset_overview_genes.txt", sep = "\t", header = T)
df_oxtp <- dataset[dataset$pathway == "OXTP",]
ggplot(df_oxtp, aes(x=nucl_div, y=dNdS_FEL)) +
  geom_point()
# Corelation
oxtp_cor <- cor.test(df_oxtp$nucl_div, df_oxtp$dNdS_FEL, method = 'spearman')
print(oxtp_cor)

# Corelation test for the other pathways (all individually)
df_dopp <- dataset[dataset$pathway == "DOPP",]
df_serp <- dataset[dataset$pathway == "SERP",]
df_step <- dataset[dataset$pathway == "STEP",]
df_vtr <- dataset[dataset$pathway == "VT-R",]

dopp_cor <- cor.test(df_dopp$nucl_div, df_dopp$dNdS_FEL, method = 'spearman')
serp_cor <- cor.test(df_serp$nucl_div, df_serp$dNdS_FEL, method = 'spearman')
step_cor <- cor.test(df_step$nucl_div, df_step$dNdS_FEL, method = 'spearman')
vtr_cor <- cor.test(df_vtr$nucl_div, df_vtr$dNdS_FEL, method = 'spearman')

print(dopp_cor)
print(serp_cor)
print(step_cor)
print(vtr_cor)
```
All tests provide statistical significance for corelation between (π) and (ω) except for the VT-R. The latter has only 7 points and the inferred rho is negative, as opposed to the rest. This can be due to the low number of observations.



shapiro.test(resid(<model>)) will tell if residuals follow normal distribution. If p > 0.05, residuals follow normal distribution

